supabase/functions/on_channel_update/index.ts

```diff
@@ -27,7 +27,31 @@ serve(async (event: Request) => {
     const record = body.record
     console.log('record', record)
 
-    if (record.public) {
+    if (record.public && record.ios) {
+      // find all other channels with same app_i with ios true and update them to false
+      const { error } = await supabaseAdmin()
+        .from('channels')
+        .update({ public: false })
+        .eq('app_id', record.app_id)
+        .eq('ios', true)
+        .neq('id', record.id)
+      if (error)
+        console.log('error', error)
+    }
+
+    if (record.public && record.android) {
+      // find all other channels with same app_i with android true and update them to false
+      const { error } = await supabaseAdmin()
+        .from('channels')
+        .update({ public: false })
+        .eq('app_id', record.app_id)
+        .eq('android', true)
+        .neq('id', record.id)
```

Comment:
since this query is used 3 times in this code, you could pre create it and save the query in a variable and then await for the values. 

something like:
```
      const updateChannels = supabaseAdmin()
        .from('channels')
        .update({ public: false })
        .eq('app_id', record.app_id)
        .neq('id', record.id)
```

and then based on the conditional add one more filter 
```
      const { error } = await updateChannels
        .eq('android', true)
```